apps/ios/SimpleX SE/ShareModel.swift

```diff
@@ -59,9 +67,17 @@ class ShareModel: ObservableObject {
         Task {
             switch await self.sendMessage() {
             case let .success(item):
-                await MainActor.run { self.bottomBar = .loadingBar(progress: .zero) }
+                if item.chatInfo.chatType != .group {
+                    await MainActor.run {
+                        self.bottomBar = .loadingBar(progress: .zero)
+                    }
+                }
                 // Listen to the event loop for progress events
-                EventLoop.shared.set(itemId: item.chatItem.id, model: self)
+                EventLoop.shared.set(
```

Comment:
On fast networks we may have a race condition here that event would arrive before sendMessage returns (particularly when sending small file to a large group it is very likely to happen on any network), so event loop has to be fully configured for processing before calling sendMessage.

One option could be to stop and create the new loop on the next invocation. But it may result in the process being suspended before stopping the loop, so the next invocation needs to be able to stop the old loop too - probably it has to be tracked in some global variable...